### PR TITLE
watch: fix missing events from symlinked path (e.g. tmp on macOS)

### LINKED ISSUE #10513: compose watch can't run from a /tmp on OSX
### Description

running `compose watch` from `/tmp` on OSX won't detect file updates.
the root cause is that this is actually a symlink to /private/tmp

see https://twitter.com/aurelien_lair/status/1652309539188875264

### Steps To Reproduce

_No response_

### Compose Version

_No response_

### Docker Environment

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by milas**
**What I did**
Fix for watching for files in paths that have symlinks in them.

Resolve the project directory before joining it to the trigger path. Additionally, resolve symlinks in any absolute trigger paths.

There's probably (definitely) still some edge cases in here, but this should fix the most common case, which is temporary directories on macOS, which are symlinked to their real location under `/private`.

**Related issue**
* Fixes #10513 

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**
camouflaged owl

### CODE REVIEW

File: pkg/compose/watch.go [hunk 1]
```diff
@@ -249,9 +249,19 @@ func loadDevelopmentConfig(service types.ServiceConfig, project *types.Project)
 	if err != nil {
 		return nil, err
 	}
+	// TODO: compose-go should return the canonical path with all symlinks resolved
```
Comment (ndeloof) [inline pkg/compose/watch.go:252]: indeed.
As we can release compose-go any time (we use to just depend on HEAD, but _some_ requested we adopt versions) would it make better sense to fix this in `loader.absPath` and by the way make this method public so we can use it in watch ?

**Review by ndeloof — APPROVED** [review]
LGTM but I'd prefer we fix this in compose-go with a patch release

**Comment by milas** [comment]
Switched this to draft - going to make the changes in `compose-go` as suggested and will likely end up closing this PR in favor of that

**Comment by milas** [comment]
Closing in favor of:
 * compose-spec/compose-go#436

**Pull request closed without merging**